python/cugraph/structure/number_map.py

```diff
@@ -88,7 +88,7 @@ def __init__(self, df, src_col_names, dst_col_names, id_type,
                     self.df[newname] = tmp[newname].append(tmp_dst[oldname])
                 self.df['count'] = tmp['count'].append(tmp_dst['count'])
             else:
-                for newname, oldname in zip(self.col_names, dst_col_names):
+                for newname in self.col_names:
```

Comment:
It might be good to add a unit test for number_map that exercises this functionality (a simple test using reproducer from #1316)

Reply:
added